Library/Homebrew/cmd/outdated.rb

```diff
@@ -37,4 +33,30 @@ def outdated_brews(formulae)
       end
     end.compact
   end
+
+  def formulae_to_check
+    ARGV.resolved_formulae.any? ? ARGV.resolved_formulae : Formula.installed
+  end
+
+  def print_outdated
+    outdated_brews(formulae_to_check) do |f, versions|
+      if ($stdout.tty? || ARGV.verbose?) && !ARGV.flag?("--quiet")
+        puts "#{f.full_name} (#{versions*', '} < #{f.pkg_version})"
+      else
+        puts f.full_name
+      end
+    end
+  end
+
+  def print_outdated_json
+    json = []
+    outdated = outdated_brews(formulae_to_check) do |f, versions|
+      json << {:name => f.name,
+               :installed_versions => versions.collect(&:to_s),
```

Comment:
My one thought here:

`installed_versions` is really "installed versions except current version". Maybe that should be:

```ruby
:installed_versions => versions.collect(&:to_s) + f.pkg_version.to_s
```

To make that list whole?

Or, make the key `other_installed_versions`?